cpp/src/arrow/dataset/file_csv.cc

```diff
@@ -503,5 +507,31 @@ Future<> CsvFileWriter::FinishInternal() {
   return Status::OK();
 }
 
+Result<std::shared_ptr<FragmentScanOptions>> CsvFragmentScanOptions::from(
+    const std::unordered_map<std::string, std::string>& configs) {
+  std::shared_ptr<CsvFragmentScanOptions> options =
+      std::make_shared<CsvFragmentScanOptions>();
+  for (auto const& it : configs) {
+    auto& key = it.first;
+    auto& value = it.second;
+    if (key == "delimiter") {
+      options->parse_options.delimiter = value.data()[0];
+    } else if (key == "quoting") {
+      options->parse_options.quoting = parseBool(value);
+    } else if (key == "column_type") {
+      int64_t schema_address = std::stol(value);
```

Comment:
should we check for possible `-1` ?

Reply:
I changed it in Java side to not add invalid schema address